java-frontend/src/main/java/org/sonar/java/se/checks/FlowComputation.java

```diff
@@ -117,14 +122,51 @@ private void run(@Nullable final ExplodedGraph.Node currentNode, @Nullable final
     final ExplodedGraph.Node parent = currentNode.parent();
     learnedConstraints.stream()
       .filter(addToFlow.and(Objects::nonNull))
-      .forEach(lc -> flow.add(location(parent, learnedConstraintMessage(lc, parent))));
+      .forEach(lc -> {
+        flow.add(location(parent, learnedConstraintMessage(lc, currentNode, parent)));
+        if (parent.programPoint.syntaxTree().is(Tree.Kind.METHOD_INVOCATION)) {
+          List<MethodYield> usedMethodYields = currentNode.usedMethodYields(parent);
+          if (!usedMethodYields.isEmpty()) {
+            flow.addAll(flowFromMethodInvocation(usedMethodYields, parent));
+          }
+        }
+      });
+
     return learnedConstraints;
   }
 
-  private static String learnedConstraintMessage(Constraint lc, ExplodedGraph.Node currentNode) {
-    Tree nodeTree = currentNode.programPoint.syntaxTree();
+  private List<JavaFileScannerContext.Location> flowFromMethodInvocation(List<MethodYield> usedMethodYields, ExplodedGraph.Node parentNode) {
+    // TODO use other yields as alternative flow?
+    MethodYield firstMethodYield = usedMethodYields.get(0);
+    int argumentIndex = correspondingArgumentIndex(symbolicValue, parentNode);
+    return firstMethodYield.flow(argumentIndex);
+  }
+
+  private static int correspondingArgumentIndex(SymbolicValue candidate, ExplodedGraph.Node invocationNode) {
+    MethodInvocationTree mit = (MethodInvocationTree) invocationNode.programPoint.syntaxTree();
+    List<SymbolicValue> arguments = argumentsUsedForMethodInvocation(invocationNode, mit);
+    return arguments.indexOf(candidate);
+  }
+
+  private static List<SymbolicValue> argumentsUsedForMethodInvocation(ExplodedGraph.Node invocationNode, MethodInvocationTree mit) {
+    List<SymbolicValue> values = new ArrayList<>(invocationNode.programState.peekValues(mit.arguments().size()));
```

Comment:
This is not properly tested : you only have invocations with one argument in your tests. AFAIK (pblm raised by tibor) there might be an issue with indexes when you are in the case of exceptions (because in execptional path return value is on the stack whereas otherwise it is not). So I would add more complex cases to be 100% sure things are working properly and/or document if it works only on the context of exceptional paths.